pkg/asset/installconfig/powervs/metadata.go

```diff
@@ -240,3 +241,40 @@ func (m *Metadata) IsVPCPermittedNetwork(ctx context.Context, vpcName string, ba
 
 	return false, nil
 }
+
+// GetSubnetID gets the ID of a VPC subnet by name and region.
+func (m *Metadata) GetSubnetID(ctx context.Context, subnetName string, vpcRegion string) (string, error) {
+	subnet, err := m.client.GetSubnetByName(ctx, subnetName, vpcRegion)
+	if err != nil {
+		return "", err
+	}
+	return *subnet.ID, err
+}
+
+// GetVPCSubnets gets a list of subnets in a VPC.
+func (m *Metadata) GetVPCSubnets(ctx context.Context, vpcName string) ([]vpcv1.Subnet, error) {
+	vpc, err := m.client.GetVPCByName(ctx, vpcName)
+	if err != nil {
+		return nil, err
+	}
+	subnets, err := m.client.GetVPCSubnets(ctx, *vpc.ID)
+	return subnets, err
```

Comment:
```suggestion
	return m.client.GetVPCSubnets(ctx, *vpc.ID)
```

Reply:
Forgot to check if `err != nil` etc. Added that instead!